durable-storage/src/merkle_layer/tree.rs

```diff
@@ -46,3 +46,190 @@ impl Avl {
         set(&mut self.root, key, data);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::BTreeMap;
+
+    use bytes::Bytes;
+    use proptest::prelude::*;
+
+    use super::*;
+    use crate::merkle_layer::KEY_MAX_SIZE;
+    use crate::merkle_layer::Key;
+
+    #[derive(Debug, Clone)]
+    enum Operation {
+        Get(Key),
+        Upsert(Key, Bytes),
+        Delete(Key),
+    }
+
+    fn key_strategy() -> impl Strategy<Value = Key> {
+        (1usize..=KEY_MAX_SIZE)
```

Comment:
```suggestion
        (1..=KEY_MAX_SIZE)
```
mega nit - the type is unnecessary here since `KEY_MAX_SIZE` is `usize`